libcxx/include/__algorithm/make_heap.h

```diff
@@ -31,13 +33,21 @@ inline _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX14 void
 __make_heap(_RandomAccessIterator __first, _RandomAccessIterator __last, _Compare&& __comp) {
   __comp_ref_type<_Compare> __comp_ref = __comp;
 
-  using difference_type = typename iterator_traits<_RandomAccessIterator>::difference_type;
-  difference_type __n   = __last - __first;
+  using __diff_t = __iter_diff_t<_RandomAccessIterator>;
+  const __diff_t __n   = __last - __first;
+
+  static const bool __assume_both_children = is_arithmetic<__iter_value_type<_RandomAccessIterator> >::value;
```

Comment:
```suggestion
  // While it would be correct to always assume we have both children, in practice we
  // observed this to be a performance improvement only for arithmetic types.
  static const bool __assume_both_children = is_arithmetic<__iter_value_type<_RandomAccessIterator> >::value;
```